File: 백준/Bronze/1434. 책 정리/책 정리.js
const input = require('fs').readFileSync(0).toString().trim().split('\n');
const [N, M] = input[0].split(' ').map(Number);
const boxes = input[1].split(' ').map(Number);
const books = input[2].split(' ').map(Number);

let boxIdx = 0;
let bookIdx = 0;

while (boxIdx < N && bookIdx < M) {
    if (boxes[boxIdx] >= books[bookIdx]) {
        boxes[boxIdx] -= books[bookIdx];
        bookIdx++;
    } else {
        boxIdx++;
    }
}

console.log(boxes.reduce((acc, cur) => acc + cur, 0));

File: 백준/Bronze/1434. 책 정리/책 정리.py
N, M = map(int, input().split())
boxes = list(map(int, input().split()))
books = list(map(int, input().split()))

box_idx = 0
book_idx = 0

while box_idx < N and book_idx < M:
    if boxes[box_idx] >= books[book_idx]:
        boxes[box_idx] -= books[book_idx]
        book_idx += 1
    else:
        box_idx += 1

print(sum(boxes))